### ISSUE TITLE
documentation for inline disabling is deceptive

### ISSUE THREAD

**Issue by jeffreybiles** [bbatsov/rubocop#1625]
Documentation states the following

"""
One or more cops can be disabled on a single line with an end-of-line comment.

`for x in (0..19) # rubocop:disable Style/AvoidFor`
"""

This implies that the cops are only disabled on that line, and not on the following lines.  However, I have confirmed that cops continue to be disabled.  The following example passes rubocop.

```
    return unless @user = @job.try(:client).try(:user) # rubocop:disable AssignmentInCondition
    return unless @user = @job.try(:client).try(:user)
```

A suggested change might be

"""
One or more cops can be disabled with an end-of-line comment.

`for x in (0..19) # rubocop:disable Style/AvoidFor`

However, you must still reenable the cops in the following line.
"""

**Comment by jonas054** [comment]
The documentation describes the intended functionality. It also describes the functionality that I get when I run rubocop 0.28.0 with default configuration.

I'm not sure why you don't get an offense report for the second line. It could be that you have disabled `AssignmentInCondition` somewhere else. Either in `.rubocop.yml` or in a comment earlier in the inspected file.

**Comment by jeffreybiles** [comment]
It appears the project is running rubocop 0.18.1.  That is likely the
issue. Sorry for the false alarm.

**Comment by jeffkowalski** [comment]
I'm still seeing a problem, using 0.78.0:
```
$ cat test.rb 
# coding: utf-8     # rubocop:disable Style/Encoding
# frozen_string_literal: true

$ rubocop --version
0.78.0

$ rubocop test.rb
Inspecting 1 file
W

Offenses:

test.rb:1:1: W: Lint/MissingCopEnableDirective: Re-enable Style/Encoding cop with # rubocop:enable after disabling it.
# coding: utf-8     # rubocop:disable Style/Encoding
^

1 file inspected, 1 offense detected
```

**Comment by jonas054** [comment]
End-of-line disable comments can't be used to disable an offense in a comment, currently. It will work like the other kind of disable comment, the one that needs an enable. It's an edge case, but it seems like a bug to me.

@jeffkowalski Please open a new issue. If you mention me in it, I will try to fix the problem.
